x-pack/plugins/cases/server/services/cases/transform.ts

```diff
@@ -0,0 +1,240 @@
+): SavedObjectsBulkUpdateResponse<CaseAttributes> {
+  return {
+    ...response,
+    saved_objects: response.saved_objects.map((so) => ({
+      ...so,
+      ...transformUpdateResponseToExternalModel(so),
+    })),
+  };
+}
+
+export function transformUpdateResponseToExternalModel(
+  updatedCase: SavedObjectsUpdateResponse<ESCaseAttributes>
+): SavedObjectsUpdateResponse<CaseAttributes> {
+  const { connector, external_service, ...restUpdateAttributes } = updatedCase.attributes ?? {};
+
+  const transformedConnector = transformESConnectorToExternalModel({
+    // if the saved object had an error the attributes field will not exist
+    connector,
+    references: updatedCase.references,
+    referenceName: connectorIdReferenceName,
+  });
+
+  let externalService: CaseFullExternalService | null | undefined;
+
+  // if external_service is not defined then we don't want to include it in the response since it wasn't passed it as an
+  // attribute to update
+  if (external_service !== undefined) {
+    externalService = transformESExternalService(external_service, updatedCase.references);
+  }
+
+  return {
+    ...updatedCase,
+    attributes: {
+      ...restUpdateAttributes,
+      ...(transformedConnector && { connector: transformedConnector }),
+      // if externalService is null that means we intentionally updated it to null within ES so return that as a valid value
+      ...(externalService !== undefined && { external_service: externalService }),
+    },
+  };
+}
+
+export function transformAttributesToESModel(
+  caseAttributes: CaseAttributes
+): {
+  attributes: ESCaseAttributes;
+  references?: CaseSavedObjectReference[];
+};
+export function transformAttributesToESModel(
+  caseAttributes: Partial<CaseAttributes>
+): {
+  attributes: Partial<ESCaseAttributes>;
+  references?: CaseSavedObjectReference[];
+};
+export function transformAttributesToESModel(
+  caseAttributes: Partial<CaseAttributes>
+): {
+  attributes: Partial<ESCaseAttributes>;
+  references?: CaseSavedObjectReference[];
+} {
+  const { connector, external_service, ...restAttributes } = caseAttributes;
+
+  let transformedAttributes: Partial<ESCaseAttributes> = { ...restAttributes };
+  let pushConnectorId: string | undefined | null;
+
+  if (external_service) {
+    let restExternalService: ExternalServicesWithoutConnectorId | null | undefined;
+    ({ connector_id: pushConnectorId, ...restExternalService } = external_service);
+    transformedAttributes = {
+      ...transformedAttributes,
+      external_service: restExternalService,
+    };
+  } else if (external_service === null) {
+    transformedAttributes = { ...transformedAttributes, external_service: null };
+  }
+
+  if (connector) {
+    transformedAttributes = {
+      ...transformedAttributes,
+      connector: {
+        name: connector.name,
+        type: connector.type,
+        fields: transformFieldsToESModel(connector),
+      },
+    };
+  }
```

Comment:
I cannot think of a way to remove the casting (`restExternalService as ExternalServicesWithoutConnectorId
`). TS won this fight 😄 .

```suggestion
  const { connector, external_service, ...restAttributes } = caseAttributes;
  const { connector_id: pushConnectorId, ...restExternalService } = external_service ?? {};

  const transformedAttributes = {
    ...restAttributes,
    ...(external_service != null
      ? {
          external_service: restExternalService as ExternalServicesWithoutConnectorId,
        }
      : { external_service: null }),
    ...(connector != null
      ? {
          connector: {
            name: connector.name,
            type: connector.type,
            fields: transformFieldsToESModel(connector),
          },
        }
      : {}),
  };
```

Reply:
Haha, thanks, I took another stab at this let me know what you think.